# BAR CLAMP
spawn_mesh: true
object_name: "object"
object_mesh_file: "package://ros2_transformers/meshes/objects/bar_clamp.dae"
object_dimensions: [0.06, 0.12, 0.04]  # [length, width, height]
object_pose: [0.5, -0.15, 0.1, 0, 0, 0]

# STRAWBERRY
# spawn_mesh: true
# object_name: "object"
# object_mesh_file: "package://ros2_transformers/meshes/objects/strawberry.dae"
# object_dimensions: [0.02]  # [radius]
# object_pose: [0.5, 0, 0., 0, 0, 0]

# Total planning attempts
max_solutions: 10
execute: true
action_name: "sample_grasp_poses"
# move_group:
#   capabilities: "move_group/ExecuteTaskSolutionCapability"

# Planning group and link names
arm_group_name: "interbotix_arm"
eef_name: "interbotix_gripper"
hand_group_name: "interbotix_gripper"
hand_frame: "locobot_fingers_link"

# Poses
hand_open_pose: "Released"
hand_close_pose: "Grasping"
arm_home_pose: "Sleep"
deep_grasps: false
# Scene frames
world_frame: "world"
table_reference_frame: "world"
object_reference_frame: "world"
camera_reference_frame: "world"
bowl_reference_frame: "world"
surface_link: "table"

spawn_objs: ["barclamp"]

spawn_objs: ["barclamp"]

barclamp_name: "barclamp"
barclamp_dimensions: [0.08, 0.12, 0.08]
barclamp_pose: [0.4, -0.15, 0.03, 0, 0, 0]   # [height, radius]
barclamp_place_pose: [0.4, 0.1,  0.1, 0, 0, 0]

# Table model
spawn_table: true
table_name: "table"
table_dimensions: [0.4, 0.5, 0.00002]  # [length, width, height]
table_pose: [0.4, 0,0.00001, 0, 0, 0]
# table_pose: [0.5, 0, 0, 0, 0, 0]

# Gripper grasp frame transform [x,y,z,r,p,y]
# grasp_frame_transform: [0.04, 0, 0, 0, -1.57, 0]
grasp_frame_transform: [0.04, 0, 0, 0, -1.57, 0]
# Place pose [x,y,z,r,p,y]
# place_pose: [0.5, 0.3, 0, 0, 0, 0]

place_surface_offset: 0.005 # place offset from table

# Valid distance range when approaching an object for picking
approach_object_min_dist: 0.05
approach_object_max_dist: 0.1

lower_object_min_dist: 0.05
lower_object_max_dist: 0.1

# Valid height range when lifting an object after pick
lift_object_min_dist: 0.03
lift_object_max_dist: 0.1
